test/foundry/Regular_Tests/Setup.sol

```diff
@@ -0,0 +1,122 @@
+    // Users
+    address user1 = address(7);
+    address user2 = address(8);
+    address user3 = address(9);
+    // Handy Variables
+    uint256 maxSupply = 10000000000 ether;
+    // Code below is executed prior to every test
+    function setUp() public {
+        // Create new instance of the treasury
+        treasury = new Treasury();
+        // Add each reward token to an array
+        rewardTokens.push(address(aurora));
+        rewardTokens.push(address(rewardToken1));
+        rewardTokens.push(address(rewardToken2));
+        rewardTokens.push(address(rewardToken3));
+        // Initialize treasury + jetStaking
+        treasury.initialize(rewardTokens, flags);
+        jetStaking = new JetStakingV1();
+        uint256 startTime = block.timestamp + 10;
+        // Schedule time for aurora stream
+        scheduleTimes.push(startTime);
+        scheduleTimes.push(startTime + 1 * oneYear);
+        scheduleTimes.push(startTime + 2 * oneYear);
+        scheduleTimes.push(startTime + 3 * oneYear);
+        scheduleTimes.push(startTime + 4 * oneYear);
+        // Schedule rewards for aurora stream
+        scheduleRewards.push(200000000 ether);
+        scheduleRewards.push(100000000 ether);
+        scheduleRewards.push(50000000 ether);
+        scheduleRewards.push(25000000 ether);
+        scheduleRewards.push(0 ether);
+        // Initializing jetStaking
+        jetStaking.initialize(
+            address(aurora), 
+            streamOwner, 
+            scheduleTimes, 
+            scheduleRewards, 
+            tauPerStream, 
+            flags, 
+            address(treasury), 
+            maxWeight, 
+            minWeight);
+        // 'Deal' - cheatcode provided by Foundry to set the balance of a token of a user
+        deal(address(aurora), address(this), maxSupply); 
+        deal(address(aurora), address(treasury), maxSupply); 
+        deal(address(rewardToken1), address(treasury), 100000000000 ether); 
+        deal(address(rewardToken2), address(treasury), 100000000000 ether); 
+        deal(address(rewardToken3), address(treasury), 100000000000 ether); 
+        // Fetching all roles from jetStaking
+        bytes32 defaultAdminRole = jetStaking.DEFAULT_ADMIN_ROLE();
+        bytes32 claimRole = jetStaking.CLAIM_ROLE();
+        bytes32 airdropRole = jetStaking.AIRDROP_ROLE();
+        bytes32 pauseRole = jetStaking.PAUSE_ROLE();
+        bytes32 streamManagerRole = jetStaking.STREAM_MANAGER_ROLE();
+        // Granting all role access to address(this)
+        jetStaking.grantRole(claimRole, address(this));
+        jetStaking.grantRole(airdropRole, address(this));
+        jetStaking.grantRole(pauseRole, address(this));
+        jetStaking.grantRole(streamManagerRole, address(this));
+        // Grant default adming role to jetStaking address
+        treasury.grantRole(defaultAdminRole, address(jetStaking));
+        vm.startPrank(address(treasury));
+        aurora.approve(address(jetStaking), 2**256 - 1);
```

Comment:
`2**256 - 1` avoid magic numbers. Please assign this value to a constant state variable (e.g `MAX_INT`) or just use the `maxSupply` if it is enough.